petstore/src/test/scala/io/finch/petstore/PetSpec.scala

```diff
@@ -0,0 +1,96 @@
+package io.finch.petstore
+
+import _root_.argonaut._
+import _root_.argonaut.Argonaut._
+import org.scalatest.prop.Checkers
+import org.scalacheck.Prop.BooleanOperators
+import org.scalatest.{FlatSpec, Matchers}
+//import org.scalatest.prop.Checkers
+
+/*
+Tests for the Pet class
+ */
+class PetSpec extends FlatSpec with Matchers with Checkers {
+
+  "The Pet codec" should "correctly decode JSON to Pet" in {
+    check{ pet: Pet =>
+      val json = pet.asJson.toString
+      Parse.decodeOption[Pet](json) === Some(pet)
+    }
+  }
+
+  it should "correctly encode a Pet to JSON" in {
+    check{ pet: Pet =>
+      Parse.decodeOption[Pet](pet.asJson.nospaces) === Some(pet)
+    }
+  }
+}
+
+/*
+Tests for the Status class
+ */
+class StatusSpec extends FlatSpec with Matchers with Checkers {
+
+  "The Status codec" should "correctly fail to decode irrelevant JSON" in {
```

Comment:
It would be better to break this into two sections, since the first three lines are testing the success cases, not failure.